### PR TITLE
Kendall tests branch

### PR DESCRIPTION
**Pull request by kendallzhu**
wrote tests for robot_commands

### CODE REVIEW

**Review by eda214 — CHANGES_REQUESTED** [review]
.emacs-buffers should be removed

File: comms/tests/test_robot_commands.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        mock_position = np.array([x, 0, angle])
+        rc.derive_speeds(mock_position)
+        assert rc._x == rc.ROBOT_MAX_SPEED
+    # extra test for speed limit
+    low_speed_limit = 100
+    rc.set_speed_limit(low_speed_limit)
+    rc.derive_speeds(mock_position)
+    assert rc._x == low_speed_limit
+    
+def test_set_waypoints():
```
Comment (eda214) [inline comms/tests/test_robot_commands.py]: Document what cases these functions are testing and what is required to pass

**Comment by wfus** [comment]
@kendallzhu

File: comms/tests/test_robot_commands.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        serialized_cmd = rc.get_serialized_command(0)
+        deserialized_cmd = rc.deserialize_command(serialized_cmd)
+        round_command(deserialized_cmd)
+        round_command(correct_cmd)
+        assert correct_cmd == deserialized_cmd
+    assert_serialize(False, False, False, 0, 0, 0)
+    assert_serialize(True, True, True, 100, 100, 2)
+    assert_serialize(False, True, False, -20, 300, -2)
+    assert_serialize(False, True, False, -200, 500, 0)
+
```
Comment (wfus) [inline comms/tests/test_robot_commands.py]: Need two lines between functions (Linting will be added soon)

File: comms/tests/test_robot_commands.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    assert_conversion(-math.pi, np.array([-200, 50]))
+    assert_conversion(-math.pi + 1, np.array([-200, -150]))
+    # also check a specific case
+    ftr = rc.field_to_robot_perspective(0, np.array([100, 0]))
+    assert np.allclose(ftr, np.array([0, 100]))
+    ftr = rc.field_to_robot_perspective(math.pi / 2, np.array([100, 0]))
+    assert np.allclose(ftr, np.array([100, 0]))
+    ftr = rc.field_to_robot_perspective(math.pi / 2, np.array([100, 100]))
+    assert np.allclose(ftr, np.array([100, 100]))
+
```
Comment (wfus) [inline comms/tests/test_robot_commands.py]: again, two lines between funcs

**Review by wfus — CHANGES_REQUESTED** [review]

**Review by eda214 — APPROVED** [review]
Documentation is good now, linting fixes may still be necessary

**Comment by wfus** [comment]
@kendallzhu  plz merge in master so linting gets checked and it's mergable

**Review by wfus — COMMENTED** [review]
Needs to merge in master and fix linting

**Comment by eda214** [comment]
This looks good to me now! Will let @wfus confirm this time just to make sure I'm not missing anything.

**Comment by wfus** [comment]
lgtm

**Review by wfus — APPROVED** [review]

**Pull request merged**